### PR TITLE
Updates mainly for Tau Embedding

### PR DESCRIPTION
**Pull request by harrypuuter**
This PR contains loads of changes needed for Tau Embedding in UL, still a draft since more changes will come

### CODE REVIEW

File: producers/photons.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from ..quantities import output as q
+from ..quantities import nanoAOD as nanoAOD
+from code_generation.producer import Producer, ProducerGroup
+
+####################
+# Set of producers used for loosest selection of electrons
```
Comment (nshadskiy) [inline producers/photons.py:6]: typo "of electrons" -> "of photons"

File: tau_embedding_settings.py [hunk 2]
```diff
@@ -181,4 +194,823 @@ def setup_embedding(configuration: Configuration, scopes: List[str]):
         ),
     )
 
+    # For the tau related triggers, in embedding, we cannot use a trigger path directly, since the are not
```
Comment (nshadskiy) [inline tau_embedding_settings.py:197]: typo: "they"

File: tau_embedding_settings.py [hunk 3]
```diff
@@ -181,4 +194,823 @@ def setup_embedding(configuration: Configuration, scopes: List[str]):
...
+    # correctly represented in embedded samples. Instead, it is possible to match to an earier filter
```
Comment (nshadskiy) [inline tau_embedding_settings.py:198]: typo: "earlier"?

File: tau_embedding_settings.py [hunk 4]
```diff
@@ -181,4 +194,823 @@ def setup_embedding(configuration: Configuration, scopes: List[str]):
...
+                            "p1_ptcut": 40,
+                            "p2_ptcut": 40,
+                            "p1_etacut": 2.1,
+                            "p2_etacut": 2.1,
+                            "p1_filterbit": 6,
+                            "p1_trigger_particle_id": 15,
+                            "p2_filterbit": 6,
+                            "p2_trigger_particle_id": 15,
+                            "max_deltaR_triggermatch": 0.4,
+                        },
```
Comment (nshadskiy) [inline tau_embedding_settings.py:272]: The filterbits are only changed for 2018. Is this intended? (sane in mt)

File: tau_embedding_settings.py [hunk 5]
```diff
@@ -181,4 +194,823 @@ def setup_embedding(configuration: Configuration, scopes: List[str]):
...
+                }
+                for wp in [
+                    # "VVVLoose",
+                    # "VVLoose",
+                    # "VLoose",
+                    # "Loose",
+                    # "Medium",
+                    "Tight",
+                    # "VTight",
+                    # "VVTight",
```
Comment (nshadskiy) [inline tau_embedding_settings.py:479]: Does this mean that there is only the tight WP or that the others are not needed?

**Review by nshadskiy — APPROVED** [review]
Looks good to me!

File: tau_embedding_settings.py [hunk 6]
```diff
@@ -55,44 +63,51 @@ def setup_embedding(configuration: Configuration, scopes: List[str]):
...
                     "embedding_trigger_sf": "Trg_IsoMu24_pt_eta_bins",
+                    "muon_trg_extrapolation": 1.0,
                 },
                 {
                     "flagname": "trg_wgt_single_mu27",
                     "embedding_trigger_sf": "Trg_IsoMu27_pt_eta_bins",
+                    "muon_trg_extrapolation": 1.0,
                 },
                 {
                     "flagname": "trg_wgtsingle_mu24Ormu27",
```
Comment (nshadskiy) [inline tau_embedding_settings.py:80]: typo: wgt`_`single

File: tau_embedding_settings.py (hunk 4 above)
Comment (harrypuuter) [reply tau_embedding_settings.py:272]: I focused on 22018 only for now, should be checked in 2017 and 2016

**Pull request merged**